zerver/tests/test_auth_backends.py

```diff
@@ -2828,6 +2829,125 @@ def test_social_auth_no_key(self) -> None:
+
+    def social_auth_test(
+        self,
+        *args: Any,
+        **kwargs: Any,
+    ) -> HttpResponse:
+        idp_discovery_endpoint_payload_dict = {
+            "issuer": self.BASE_OIDC_URL,
+            "authorization_endpoint": self.AUTHORIZATION_URL,
+            "token_endpoint": self.ACCESS_TOKEN_URL,
+            "userinfo_endpoint": self.USER_INFO_URL,
+            "response_types_supported": [
+                "code",
+                "id_token",
+                "id_token token",
+                "code token",
+                "code id_token",
+                "code id_token token",
+            ],
+            "jwks_uri": self.JWKS_URL,
+            "id_token_signing_alg_values_supported": ["HS256", "RS256"],
+            "subject_types_supported": ["public"],
+            "token_endpoint_auth_methods_supported": ["client_secret_post", "client_secret_basic"],
+        }
+
+        # We need to run the social_auth_test procedure with a mock response set up for the
+        # OIDC discovery endpoint as that's the first thing requested by the server when a user
+        # starts trying to authenticate.
+        with responses.RequestsMock(assert_all_requests_are_fired=False) as requests_mock:
+            requests_mock.add(
+                requests_mock.GET,
+                f"{self.BASE_OIDC_URL}/.well-known/openid-configuration",
+                match_querystring=False,
+                status=200,
+                body=json.dumps(idp_discovery_endpoint_payload_dict),
+            )
+            result = super().social_auth_test(*args, **kwargs)
+
+        return result
+
+    def social_auth_test_finish(self, *args: Any, **kwargs: Any) -> HttpResponse:
+        # Trying to generate a (access_token, id_token) pair here in tests that would
+        # successfully pass validation by validate_and_return_id_token is impractical
+        # and unnecessaru (see upstream implementation of the method for how the validation works).
+        # We can simply mock the method to make it succeed and return an empty dict, because
+        # the return value is not used for anything.
+        with mock.patch.object(
+            GenericOpenIdConnectBackend, "validate_and_return_id_token", return_value={}
+        ):
+            return super().social_auth_test_finish(*args, **kwargs)
+
+    def register_extra_endpoints(
+        self,
+        requests_mock: responses.RequestsMock,
+        account_data_dict: Dict[str, str],
+        **extra_data: Any,
+    ) -> None:
+        requests_mock.add(
+            requests_mock.GET,
+            self.JWKS_URL,
+            status=200,
+            json=json.loads(settings.EXAMPLE_JWK),
+        )
+
+    def generate_access_url_payload(self, account_data_dict: Dict[str, str]) -> str:
+        return json.dumps(
+            {
+                "access_token": "foobar",
+                "expires_in": time.time() + 60 * 5,
+                "id_token": "foo",
```

Comment:
Maybe do `abcd1234` or something along those lines to at suggest a normal `id_token` format?